python_anvil/api.py

```diff
@@ -85,6 +85,17 @@ def generate_pdf(self, payload: Union[AnyStr, Dict]):
         else:
             raise ValueError("`payload` must be a valid JSON string or a dict")
 
+        # For HTML-type payloads, data must be a dict (or JSON object).
+        if data.type == 'html' and isinstance(data.data, list):
```

Comment:
sorta feels weird to look for a specific way the data could be wrong, instead of looking to see if the data is not right?

Like, why not check `if data.type == 'html and not isinstance(data.data, dict)` here?

I guess if you the Typing you do below in this PR ensures that it's either a `dict` or a `list`, what you have is sorta the same thing...but it feels funny...

Reply:
I see what you mean. I'll change that so it makes sense with the type being used.